### PR TITLE
Add function to get the first of the next month

### PR DESCRIPTION
**Pull request by coleaeason**
@deetergp this might help you with the thing you're doing in auth. This seems better than passing around a sqlite db object to functions that don't need it only to get the first of the next month.

cc @tylerkaraszewski

### CODE REVIEW

**Comment by coleaeason** [comment]
If you don't need it feel free to close this.

File: libstuff/STime.cpp [hunk 1]
```diff
@@ -54,3 +54,30 @@ string SCURRENT_TIMESTAMP_MS() {
     snprintf(msString, 5, "%03lu", ms);
     return timestamp + "." + msString;
 }
+
+string SFirstOfNextMonth(string timeStamp) {
+
+    list<string> parts = SParseList(timeStamp, '-');
+
+    struct tm t = {0};  // Initalize to all 0's
+    t.tm_year = stoull(parts.front()) - 1900;  // This is year - 1900
```
Comment (tylerkaraszewski) [inline libstuff/STime.cpp:63]: stoull throws, should catch (below as well)

File: libstuff/STime.cpp [hunk 2]
```diff
@@ -54,3 +54,30 @@ string SCURRENT_TIMESTAMP_MS() {
...
+    parts.pop_front();
+
+    // Month values start at 0 in tm structs, so values are off by one.
+    uint64_t month = stoull(parts.front()) -1;
```
Comment (tylerkaraszewski) [inline libstuff/STime.cpp:67]: Pretty sure that months `01` through `09` are going to be interpreted as octal if you don't specify base 10, which is great except for in August and September.

File: libstuff/STime.cpp [hunk 3]
```diff
@@ -54,3 +54,30 @@ string SCURRENT_TIMESTAMP_MS() {
     snprintf(msString, 5, "%03lu", ms);
     return timestamp + "." + msString;
 }
+
+string SFirstOfNextMonth(string timeStamp) {
```
Comment (tylerkaraszewski) [inline libstuff/STime.cpp:58]: `const string&` param.

File: libstuff/STime.cpp [hunk 4]
```diff
@@ -54,3 +54,30 @@ string SCURRENT_TIMESTAMP_MS() {
...
+        t.tm_mon = 0;
+    } else {
+        t.tm_mon = month + 1;
+    }
+    t.tm_mday = 1;
+    
+    char buf[256] = {};
+    size_t length = strftime(buf, sizeof(buf), "%Y-%m-%d", &t);
+    return string(buf, length);
+}
```
Comment (tylerkaraszewski) [inline libstuff/STime.cpp:83]: ??

File: test/tests/LibStuffTest.cpp [hunk 5]
```diff
@@ -594,4 +595,14 @@ struct LibStuff : tpunit::TestFixture {
         ASSERT_TRUE(SContains(string("asdf"), "a"));
         ASSERT_TRUE(SContains(string("asdf"), string("asd")));
     }
+
+    void testFirstOfNextMonth() {
+        string timeStamp = "2020-03-01";
+        string timeStamp2 = "2020-12-01";
+        string timeStamp3 = "2020-06-17";
```
Comment (tylerkaraszewski) [inline test/tests/LibStuffTest.cpp:602]: Add test for August or Sept (possibly octal months) and `"this is not a date"` for catching exceptions.

**Review by tylerkaraszewski — CHANGES_REQUESTED** [review]

File: libstuff/STime.cpp [hunk 6]
```diff
@@ -55,29 +55,47 @@ string SCURRENT_TIMESTAMP_MS() {
...
 
-    struct tm t = {0};  // Initalize to all 0's
-    t.tm_year = stoull(parts.front()) - 1900;  // This is year - 1900
+    // Initalize to all 0's
+    struct tm t = {0};  
+
+    try {
+        // This is year - 1900
+        t.tm_year = stoull(parts.front(), 0, 10) - 1900; 
+    } catch (const std::invalid_argument& e) {
```
Comment (tylerkaraszewski) [inline libstuff/STime.cpp:68]: Annoying, but you don't need `std::`

**Review by tylerkaraszewski — APPROVED** [review]

**Comment by deetergp** [comment]
Looks good to me!

**Pull request merged**